Newsletter: How to think like a growth engineer (#9363)
PostHog/posthog.com: contents/newsletter/think-like-a-growth-engineer.md

---
title: How to think like a growth engineer
date: 2024-09-16
author:
 - ian-vanagas
featuredImage: >-
featuredImageType: full
tags:
  - Engineering
crosspost:
  - Product engineers
  - Founders
  - Blog
---

> From the outside, it can seem like growth engineers bounce around the product aimlessly: signup flows, product onboarding, revenue reporting, and subscription tracking. You might think all this doesn't matter, but as many growth teams show, incremental improvements can lead to massive gains for the entire business.

Reviewer: ```suggestion
From the outside, it can seem as if growth engineers bounce around the product aimlessly – tweaking signup and onboarding flows, and running endless experiments, many of which fail.

It's tempting to dismiss this as busywork, but that would be a mistake. Nearly every successful company has a thriving growth team at its core, constantly driving incremental improvements that accumulate to massive gains over time.
```

Reviewer: Added an en dash because I know you love them ;)

> Growth engineers capture these gains through their unique way of thinking and working. Luckily, you don't need to go to growth engineer school to learn this yourself. If you aspire to create a successful product and business, you should care about what growth engineers do to make that happen, and this post aims to help you do just that.

Reviewer: ```suggestion
Growth engineers capture these gains through their unique way of thinking and working. And, luckily, you don't need to go to growth engineer school to learn their secrets. It all starts by learning to think like they do – here's how.
```

## 1. Become a data-driven (for real)

> Everyone claims they are data-driven, but growth engineers prove it.

Reviewer: ```suggestion
Everyone claims they are data-driven, but it's more than a meme for growth engineers.

Reviewer: Feel to scratch this.

![GROWTH ENGINEER RONNIE](https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2024_09_16_at_11_21_09_29ee4dbab8.png)

> Their work comes down to tracking and improving platform or business-level metrics. For example, our growth team cares about new revenue, expansion, signup conversion, activation, retention, and more. Their planning, prioritization, and execution are focused on improving these metrics.

Reviewer: ```suggestion
Their work comes down to tracking and improving platform or business-level metrics. Our growth team, for example, cares about new revenue, expansion, signup conversion, activation, retention, and more. Their planning, prioritization, and execution are focused on improving these metrics.
```

The tradeoff is caring less about products, features, and user requests. They aren't planning their roadmap months in advance or caring about accurate project timelines. They let the metrics be their north star and do whatever, wherever to improve it.

> **Takeaway:** Figure out the status quo for the area you work on. To do this:
> 1. Define and track your activation and retention numbers. 
> 2. Track how they move week over week. 
> 3. Track how they move with the changes you ship.
>
> You can't think like a growth engineer if you don't have a metrics baseline to start from.
>
> For more advanced users, create a dashboard using a framework like [AARRR](/product-engineers/aarrr-pirate-funnel) or [growth loops](/product-engineers/growth-loops) to monitor your key flow conversions, activation, and retention.

> > ![AARRR and growth loops](https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2024_09_17_at_11_56_15_193aeaa117.png)

Author: I will test some stuff out in Substack and see what looks best

1. **Hypothesis over requirements.** Software engineers work on features users clearly need. Growth engineers work on more unknowns and care more about exploration and discovery. To do this, they develop hypotheses and run experiments to validate their assumptions.

2. **Iteration over stability.** While most engineers focus on developing stable, bug-free code, growth engineers would rather fail fast and iterate. For example, a software engineer might feel uncomfortable adding a dependency before evaluating it, while a growth engineer will ship it to help them test faster. 

3. **Pragmatism over perfection.** Growth engineers know their experiments might fail and get removed. This means they ship the "good enough" version over the maintainable and scalable one. They know they can always improve it later.

> **Takeaway:** Instead of blindly shipping the next feature on your list, think about the hypothesis behind it.
>
> For example, adding customizable notifications will get users to return more often, improving retention. After adding the feature, use the metrics you set up in the previous section to validate your hypothesis.

## 3. Prioritize like a growth engineer

> Engineers are like bus drivers, helping move a product along its roadmap toward success. Growth engineers are like Uber drivers, bouncing from experiment to experiment stacking little wins that payoff in the long term. 

Reviewer: Just realized this would make a great visual – could we get Lottie do a couple of hogs, one driving a bus on a fixed route, another an uber? Nice to have, not must have.

3. **Create hypotheses.** Dashboard templates help teams get value out of PostHog quickly. They hypothesized that adding a dashboard template to the onboarding flow using actions would lead to better activation.

4. **Implement as small of an experiment as reasonably possible.** They added a step in the onboarding flow to choose a dashboard template and fill out the variables by creating actions.

![Dashboard templates in onboarding](https://res.cloudinary.com/dmukukwp6/image/upload/template_27fddd57a8.png)

> **Takeaway:** Now it is time to go from post-hoc analysis to proactive prioritization. With your target area, metric, and hypothesis in mind, develop a list of experiments to run that can improve your product. Think about how you can test your ideas in the smallest way possible.
>
> For example, if you are trying to improve activation by changing the onboarding flow, could you test removing steps instead of adding new ones? 

## 4. Trust the (experimentation) process

> Engineers might see the word "process" and shutter, but to be data-driven, you need to follow process to ensure you can trust your data when making decisions. When it comes to running experiments, growth engineers always make sure to have:

Reviewer: ```suggestion
Engineers might see the word "process" and shudder, but being data-driven requires an established process so you can trust the data when making decisions. When it comes to running experiments, growth engineers always make sure to have:
```
